src/main/java/team/silvertown/masil/post/repository/PostQueryRepositoryImpl.java

```diff
@@ -33,34 +36,63 @@ public class PostQueryRepositoryImpl implements PostQueryRepository {
     private final QPost post = QPost.post;
 
     @Override
-    public List<PostCursorDto> findSliceBy(User user, NormalListRequest request) {
+    public List<PostCursorDto> findScrollByAddress(User user, NormalListRequest request) {
         // TODO: 좋아요 구현 후 로그인한 사용자 본인이 좋아요한 포스트인지 쿼리 추가
-        BooleanExpression cursorCondition = getCursorFilter(request.order(), request.cursor());
-        BooleanBuilder condition = new BooleanBuilder()
+        Predicate openness = getOpenness(user, null);
+        BooleanBuilder condition = getBasicCondition(request.getScrollRequest(), openness);
+
+        condition
+            .and(post.address.depth1.eq(request.getDepth1()))
+            .and(post.address.depth2.eq(request.getDepth2()))
+            .and(post.address.depth3.eq(request.getDepth3()));
+
+        return queryPostsWith(user, condition, request.getScrollRequest());
+    }
+
+    @Override
+    public List<PostCursorDto> findScrollByUser(
+        User loginUser,
+        User author,
+        ScrollRequest request
+    ) {
+        // TODO: 좋아요 구현 후 로그인한 사용자 본인이 좋아요한 포스트인지 쿼리 추가
+        Predicate openness = getOpenness(loginUser, author);
+        BooleanBuilder condition = getBasicCondition(request, openness);
+
+        return queryPostsWith(loginUser, condition, request);
+    }
+
+    private Predicate getOpenness(User loginUser, User author) {
+        if (Objects.nonNull(loginUser) && Objects.equals(loginUser, author)) {
+            return null;
+        }
+
+        return post.isPublic.isTrue();
+    }
```

Comment:
`Objects.nonNull(loginUser)` 이 부분은 없어도 되겠네요!

```suggestion
    private Predicate getOpenness(User loginUser, User author) {
        if (Objects.equals(loginUser, author)) {
            return null;
        }

        return post.isPublic.isTrue();
    }
```

Reply:
만약 둘 다 null이면 어쩌죠.. 물론 흐름 상 그럴 일이 없긴 할텐데

Reply:
내부에 널처리가 있어서 괜찮습니다!!

<img width="869" alt="image" src="https://github.com/Team-SilverTown/Team-SilverTown-MasilGasil-BE/assets/31983961/c054e6b8-c10f-4eb5-b22a-5dd9fc995e34">

Reply:
제가 알기론 null == null은 true가 나오는거로 알고있습니다
```java
    /**
     * Returns {@code true} if the arguments are equal to each other
     * and {@code false} otherwise.
     * Consequently, if both arguments are {@code null}, {@code true}
     * is returned.  
```

Reply:
아하 둘 다 `null`일 때 `null` 리턴하면 안되는거였나요?
그럼 그대로 가시죠!